src/components/Form.js

```diff
@@ -83,11 +83,29 @@ class Form extends Component {
     this.fileUploader = React.createRef();
   }
 
+  updateNeighborhood = (lat, lng) => {
+    neighborhoodService.getNeighborhoodFor(lat, lng)
+        .then(neighborhood => this.setState((state) => {
+          // If the map coordinates have changed, we don't want to update the neighborhood with outdated info!
+          if (state.mapLat === lat && state.mapLng === lng) {
+            return {neighborhood};
+          }
+          return {};
+        }));
+  };
+
   componentDidMount = () => {
     // The neighborhood is initialized to the empty string, but we want to have a neighborhood for our
     // initial location!
-    neighborhoodService.getNeighborhoodFor(this.state.mapLat, this.state.mapLng)
-      .then(neighborhood => this.setState({neighborhood}));
+    this.updateNeighborhood(this.state.mapLat, this.state.mapLng);
+
+    // Request the user's geolocation and default to there
+    // See https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API for more information
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition((position) => {
+        this.getMapCoordinates({lat: position.coords.latitude, lng: position.coords.longitude});
+      }, (err) => console.log(err));
```

Comment:
should we handle this error?

Reply:
I'm not sure what kind of error would show up here, honestly. How do you think we should treat it?

Reply:
Not sure about the error either but if there is error in fetching the geo location of user, it would be nice to fall back to the default zoo location. What do you think?

Reply:
Sorry, for some reason I didn't see your comment!

The initial state is the zoo location. Then, if the geolocation request succeeds it updates the coordinates to be the user's current location. Thus, if the geolocation request doesn't succeed for any reason (including an error), we'll end up with the zoo location.

Long story short, I agree and that's how it's working now 🙂